GitHubRepositoryMerge.ps1

```diff
@@ -0,0 +1,131 @@
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .OUTPUTS
+        PSCustomObject
+
+    .EXAMPLE
+        Merge-GitHubRepositoryBranch -OwnerName PowerShell -RepositoryName PowerShellForGitHub -Base 'master' -Head 'new_feature' -CommitMessage 'Merging branch new_feature into master'
+#>
+
+[CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParametersetName='Elements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(ParameterSetName='Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName='Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName='Uri')]
+        [string] $Uri,
+
+        [Parameter(Mandatory)]
+        [string] $Base,
+
+        [Parameter(Mandatory)]
+        [string] $Head,
+
+        [string] $CommitMessage,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements -DisableValidation
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    $hashBody = @{
+        'base'= $Base
+        'head' = $Head
+    }
+
+    if(-not $CommitMessage -eq $null)
+    {
+        $hashBody['commit_message'] = $CommitMessage
+    }
+
+    $params = @{
+        'UriFragment' = "repos/$OwnerName/$RepositoryName/merges"
+        'Body' = (ConvertTo-Json -InputObject $hashBody)
+        'Method' = 'Post'
+        'Description' =  "Merging branch $Head to $Base in $RepositoryName"
+        'AccessToken' = $AccessToken
+        'TelemetryEventName' = $MyInvocation.MyCommand.Name
+        'TelemetryProperties' = $telemetryProperties
+        'NoStatus' = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus)
+    }
+
+    try {
+        $result = Invoke-GHRestMethod @params -ExtendedResult
+        if ($result.statusCode -eq 204)
+        {
+            Write-Log -Message "Nothing to merge. The branch $Base already contains changes from $Head" -Level Warning
+        }
+        return $result.result
+    }
+    catch {
+        #TODO: Read the error message and find out the kind of issue
```

Comment:
What's your plan here?

Reply:
The idea was to see if we could extract the status code and give a more informative message to the user. But looks like that would require parsing of the string that we get as an exception which doesn't sound clean to me. I'm leaving this as-is for now. But please do let me know your thoughts on if this can be handled better